lib/resty/rediscluster.lua

```diff
@@ -58,45 +84,42 @@ end
 _M.parse_key = parse_key
 
 
-local slot_cache = {}
-local master_nodes = {}
-
-local cmds_for_all_master = {
-    ["flushall"] = true,
-    ["flushdb"] = true
-}
-
-local cluster_invalid_cmds = {
-    ["config"] = true,
-    ["shutdown"] = true
-}
-
 local function redis_slot(str)
     return redis_crc(parse_key(str))
 end
 
+
+local function release_connection(red, config)
+    local ok,err = red:set_keepalive(config.keepalive_timeout
+            or DEFAULT_KEEPALIVE_TIMEOUT, config.keepalive_cons or DEFAULT_KEEPALIVE_CONS)
+    if not ok then
+        ngx_log(ngx_err,"set keepalive failed:", err)
+    end
+end
+
+
 local function check_auth(self, redis_client)
-    -- redis before 6.0
+    if not redis_client then
+        return nil, "redis_client can not be nil"
+    end
+
+    local count, err = redis_client:get_reused_times()
+    if not count then
+        return nil, "failed to get reused count: " .. tostring(err)
+
+    elseif count > 0 then
+        return true, nil -- reusing the connection, so already authenticated
+    end
```

Comment:
we can extract it to a function, but we can not just move it to the head of `check_auth`.

because if no auth, we don't need to call `get_reused_times`

Reply:
done